it/uk/gov/hmrc/vatregisteredcompanies/repository/PayloadBufferRepositoryISpec.scala

```diff
@@ -0,0 +1,266 @@
+import uk.gov.hmrc.vatregisteredcompanies.helpers.TestData._
+import uk.gov.hmrc.vatregisteredcompanies.repositories.PayloadWrapper
+
+class PayloadBufferRepositoryISpec extends IntegrationSpecBase {
+  override def beforeEach(): Unit = {
+    deleteAllBuffer
+  }
+
+  "Method: deleteAll" when {
+    "there are no records in the database" should {
+      "Have no records" in {
+        bufferTotalCount shouldBe 0
+        val res = payloadBufferRepository.deleteAll()
+
+        whenReady(res) { result =>
+          result shouldBe ((): Unit)
+          bufferTotalCount shouldBe 0
+        }
+      }
+    }
+
+    "there is 1 record in the database" should {
+      "Have one record" in {
+        insertOneBuffer(testPayloadCreateAndUpdates1)
+        bufferTotalCount shouldBe 1
+        val res = payloadBufferRepository.deleteAll()
+
+        whenReady(res) { result =>
+          result shouldBe ((): Unit)
+          bufferTotalCount shouldBe 0
+        }
+      }
+    }
+
+    "there are multiple records in the database" should {
+      "Have two records" in {
+        insertOneBuffer(testPayloadCreateAndUpdates)
+        insertOneBuffer(testPayloadCreateAndUpdates1)
+        bufferTotalCount shouldBe 2
+        val res = payloadBufferRepository.deleteAll()
+
+        whenReady(res) { result =>
+          result shouldBe ((): Unit)
+          bufferTotalCount shouldBe 0
+        }
+      }
+    }
+  }
+
+  "Method: insert" when {
+    "there are no records in the database" should {
+      "Add one record" in {
+        val res = payloadBufferRepository.insert(testPayloadCreateAndUpdates1)
+
+        whenReady(res) { result =>
+          result shouldBe ((): Unit)
+          bufferTotalCount shouldBe 1
+        }
+      }
+    }
+
+    "there are multiple records in the database" should {
+      "Increase the database by one record" in {
+        insertOneBuffer(testPayloadCreateAndUpdates1)
+        insertOneBuffer(testPayloadCreateAndUpdates)
+        bufferTotalCount shouldBe 2
+
+        val res = payloadBufferRepository.insert(testPayloadCreateAndUpdates1)
+
+        whenReady(res) { result =>
+          result shouldBe ((): Unit)
+          bufferTotalCount shouldBe 3
+        }
+      }
+    }
+
+    "there are multiple records in the database" should {
+      "Increase the database by two records" in {
+        insertOneBuffer(testPayloadCreateAndUpdates1)
+        insertOneBuffer(testPayloadCreateAndUpdates)
+        bufferTotalCount shouldBe 2
+
+        val res = {
+          payloadBufferRepository.insert(testPayloadCreateAndUpdates1)
+          payloadBufferRepository.insert(testPayloadCreateAndUpdates)
+        }
+
+        whenReady(res) { result =>
+          result shouldBe ((): Unit)
+          bufferTotalCount shouldBe 4
+        }
+      }
+    }
+  }
+
+  "Method: list" when {
+
+    "there are no records in the database" should {
+      "Have no records" in {
```

Comment:
"return count 0" instead?

Reply:
same changes for getOne method

Reply:
i think this should stay as "Has one record" etc as the method call list main assertion is to return the list inserted. The bufferTotalCount assertion is almost a secondary assertion.

Reply:
whatever feels right instead of Have no records" :)